### ISSUE TITLE
constants.php file always change to 0kb

### ISSUE THREAD

**Issue by slimshock** [Orangescrum/orangescrum#126]
Every now and then the file app/Config/constants.php will always change the size to 0kb. 

Its already annoying to always upload the file when this happen. 

Please advise.. thanks..

**Comment by risendev** [comment]
Seconded.  I have experienced this same issue.  Seems to occur when a new task is created.

**Comment by adilshahzad** [comment]
I do agree as well, I'm facing the same issue several time. Why its happening by the way?

**Comment by arturssmirnovs** [comment]
Yeah.. today got this issue as well.. #nobackup

**Comment by adilshahzad** [comment]
I'm having the same issue but yesterday I changed its permission to 600 & now monitoring it, but today its again changed... whats changed i don't but the configurations are fine till this time.

**Comment by slimshock** [comment]
yeah.. already tried to changed its permission to 644 or even 666  but still the file size change to 0kb.

**Comment by arturssmirnovs** [comment]
BTW.. My constants.php file emptied when i created now task..

I tried to make new task with title filled only (i think it was only filled). <-- hope this helps identify problem.

**Comment by korachoy** [comment]
i've been having the same....

**Comment by adilshahzad** [comment]
Okay here is the solution, change the file permission to 444 & you all set to go.

**Comment by mrdevnet** [comment]
@Orangescrum  do you know why?, I try to change it be config.php but it still happens.

**Comment by pprotschka** [comment]
same issue.

**Comment by NathanWarden** [comment]
I had this issue too (using the official Docker container). My workaround is just that I set the settings to what I needed, made a backup of the file, and then removed the write permissions to the file.

**Comment by InsaneSplash** [comment]
I have the same problem from time to time.....

**Comment by InsaneSplash** [comment]
To resolve this properly, I commented out line 313 in core.php which kept including the install.php file everytime a page was loaded. install.php is the file that keeps reading and writing to the constants.php file which is dumb.

app/Config/core.php
`#include("install.php");
`

**Comment by BlackMetalz** [comment]
Confirm, another solution is to remove permission of www-data or set it to read-only

**Comment by evreneter** [comment]
for v1.6.1
end of the app/Config/core.php

change 
include("setup.php");
/*include("constants.php");*/ 

to

/*include("setup.php");*/
include("constants.php");
